tests/components/stiebel_eltron/test_init.py

```diff
@@ -0,0 +1,52 @@
+"""Tests for the STIEBEL ELTRON integration."""
+
+from homeassistant.components.stiebel_eltron.const import CONF_HUB, DEFAULT_HUB, DOMAIN
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import issue_registry as ir
+from homeassistant.setup import async_setup_component
+
+
+async def test_async_setup_valid_hub(hass: HomeAssistant) -> None:
+    """Test async_setup with a valid hub."""
+    config = {
+        DOMAIN: {
+            CONF_NAME: "Stiebel Eltron",
+            CONF_HUB: DEFAULT_HUB,
+        },
+        "modbus": [
+            {
+                CONF_NAME: DEFAULT_HUB,
+                CONF_HOST: "1.1.1.1",
+            }
+        ],
+    }
+
+    assert await async_setup_component(hass, DOMAIN, config)
+    await hass.async_block_till_done(wait_background_tasks=True)
```

Comment:
Assert the issue created here also.

I don't think we need to wait for background tasks. There are no background tasks scheduled.

Reply:
We should at least test the three different code flows that creates repair issues: import success / already configured, import failure and missing config.

Reply:
Which issue should be created here?

Reply:
@MartinHjelmare so basically "import failure" is missing, as the other cases are already tested, right?

Reply:
We're missing a test for `deprecated_yaml_import_issue_cannot_connect`.